src/main/python/ayab/preferences.py

```diff
@@ -35,15 +36,37 @@
 from .machine import Machine
 from .language import Language
 from .scene import AspectRatio
+from typing import TYPE_CHECKING, Any, Callable, Literal, Optional, TypeAlias, TypeVar, TypedDict, cast 
+if TYPE_CHECKING:
+    from .ayab import GuiMain
 
+T = TypeVar('T')
 
-def str2bool(qvariant):
+def str2bool(qvariant:str|bool)->bool:
```

Comment:
dunno why you'd pass a `bool` to `str2bool` but hey. It was written to support that, I'm just noting the types.

Reply:
It's something to do with the way that the preference files are saved IIRC, which depends on the OS. Linux saves text files, Windows files are typed. This function makes the type consistent in the resulting object. Could have a better name.